tests/test_from_thread.py

```diff
@@ -524,3 +562,20 @@ async def raise_baseexception() -> None:
                     portal.call(raise_baseexception)
 
                 assert exc.value.__context__ is None
+
+    @pytest.mark.timeout(1)
+    async def test_from_async(
+        self,
+        anyio_backend_name: str,
+        portal_backend_name: str,
+        portal_backend_options: dict[str, Any],
+    ) -> None:
+        """Test that portals don't deadlock when started/used from async code."""
+
+        if anyio_backend_name == "trio" and portal_backend_name == "trio":
+            pytest.xfail("known bug (#525)")
+
+        with start_blocking_portal(
+            portal_backend_name, portal_backend_options
+        ) as portal:
+            portal.call(checkpoint)
```

Comment:
This is an annoying problem. Perhaps this could be removed from this PR, to be figured out later? I don't want `pytest-timeout` to block this from being merged.

Reply:
I think I'll just sidestep the issue of pytest-timeout for now and just rely on the timeout we have on GitHub. I want this merged for v3.7.0.

Reply:
That's just for GitHub Actions, right? Locally it would just hang forever, yes?

Reply:
Really? I haven't seen any such thing. All my searches point to `pytest-timeout` as the provider of any sort of a timeout effect.

Reply:
At any rate, if you could remove `pytest-timeout` for the time being, and make the other change I suggested, I could merge this and get it into v3.7.0. I think this is the last fix slated for that minor release.